### ISSUE TITLE
Symbols within a namespace after a heavily #ifdef class are omitted in HTML output

### ISSUE THREAD

**Issue by prince-chrismc** [doxygen/doxygen#10472]
**Describe the bug**
This is a regression as the same code worked with 1.9.5 through 1.9.7 and only gets "omitted" in 1.9.8

https://github.com/prince-chrismc/jwt-cpp/actions/runs/7119774588/job/19385610779#step:5:356

![image](https://github.com/doxygen/doxygen/assets/16867443/e03b1fda-cfc1-4e6e-b890-e5fc5cf2a829)

![image](https://github.com/doxygen/doxygen/assets/16867443/997b25db-dd8b-467f-96e6-2fa548f215ac)

Without custom CSS for sanity

![image](https://github.com/doxygen/doxygen/assets/16867443/0a188eab-e3c7-48b0-997e-85e33f99b8cb)

Ive been able to reproduce the problem using the binaries from the Github releases and well as compiled from brew on my macbook.

For sanity I tested the `GENERATE_XML` and these classes for seem to be included

```xml
<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.9.8" xml:lang="en-US">
  <compounddef id="namespacejwt_1_1algorithm" kind="namespace" language="C++">
    <compoundname>jwt::algorithm</compoundname>
    <innerclass refid="structjwt_1_1algorithm_1_1ecdsa" prot="private">jwt::algorithm::ecdsa</innerclass>
    <innerclass refid="structjwt_1_1algorithm_1_1ed25519" prot="private">jwt::algorithm::ed25519</innerclass>
    <innerclass refid="structjwt_1_1algorithm_1_1ed448" prot="private">jwt::algorithm::ed448</innerclass>
    <innerclass refid="structjwt_1_1algorithm_1_1eddsa" prot="private">jwt::algorithm::eddsa</innerclass>
    <innerclass refid="structjwt_1_1algorithm_1_1es256" prot="private">jwt::algorithm::es256</innerclass>
```

**Expected behavior**
All the structs and classes should be included

https://github.com/prince-chrismc/jwt-cpp/actions/runs/7118388391/job/19381233291#step:5:361

![image](https://github.com/doxygen/doxygen/assets/16867443/e0f6cdf2-3bca-4ae8-96db-87f1a7e2aef1)

![image](https://github.com/doxygen/doxygen/assets/16867443/03ce6ea1-be8e-4322-8358-b24bb00dda67)

**To Reproduce**
Attach a self contained example that allows us to reproduce the problem.
Such an example typically exist of some source code (can be dummy code) and a doxygen configuration file used (you can strip it using `doxygen -s -u`). After you verified the example demonstrates the problem, put it in a zip (or tarball) and attach it to the bug report. Try to avoid linking to external sources, since they might disappear in the future.

_I have no ideal whats causing the issue, i've refactored the class several time and move out/in different files to no success_

Clone and checkout https://github.com/prince-chrismc/jwt-cpp/commit/1f69de39afeb518fabc10aabeb734ff24098c825
Then it should be as easy as

```sh
cmake . -DJWT_BUILD_DOC=ON
cmake --build . --target build_docs
```

**Version**
Mention the version of doxygen used (output of `doxygen --version`) and the platform on which you run doxygen (e.g. Windows 10, 64 bit). If you run doxygen under Linux please also mention the name and version of the distribution used (output of `lsb_release -a`) and mention if you compiled doxygen yourself or that you use a binary that comes with the distribution or from the doxygen website.  

- 1.9.8

Ubunutu 22.04 (release binaries) and MacOS 14.1.1 (brew binaries)

**Stack trace**
If you encounter a crash and can build doxygen from sources yourself with debug info (`-DCMAKE_BUILD_TYPE=Debug`), a stack trace can be very helpful (especially if it is not possible to capture the problem in a small example that can be shared).

n/a?

**Additional context**

**Comment by prince-chrismc** [comment]
I had some extra free time to make a smaller repo [bug-report.tgz](https://github.com/doxygen/doxygen/files/13618936/bug-report.tgz). This one is getting it the problem, as you suggested I just kept the one file which made it much easier. Took me a while to connivence myself this was worth reporting back but I am glad it's something.

👏 

> 6d80fc7e5d03c259b1a7280972e0b28884217655 is the first bad commit 

awesome

**Comment by albert-github** [comment]
It always helps to have a smaller example.
Looks though that now there are no namespaces at all  in the results (1.9.7 and 1.9.8) version though by setting `EXTRACT_ALL = YES` the issue is shown again.

**Comment by doxygen** [comment]
I think I found the curl pit. Please check if the latest commit fixes the problem for you. Don't close the issue, this will be done automatically when the next official version is made available.

**Comment by prince-chrismc** [comment]
I finally had a chance to setup 👏 seems everything is now present

![image](https://github.com/doxygen/doxygen/assets/16867443/9894f4ff-f90a-4c08-aa6f-4e2a27252462)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
